torch/distributed/rpc/api.py

```diff
@@ -113,17 +173,22 @@ def shutdown(graceful=True):
         >>> rpc.shutdown()
     """
     global _agent
-    if _agent:
-        if graceful:
-            _wait_all_workers()
-        _destroy_rref_context(_ignore_rref_leak)
-        _agent.shutdown()
-        # clean up python rpc handler in shutdown(), see comments in
-        # PythonRpcHandler::cleanup(), call it in python API because the
-        # cleanup() function has python dependency, it assumes python
-        # interpreter exists
-        _cleanup_python_rpc_handler()
-        _agent = None
+
+    if _agent is None:
+        return
```

Comment:
We should throw an exception here saying "RpcAgent was not initialized"

Reply:
Make sense. Since we don't actually support it being called for twice if graceful is True.

Let me change.